### PR TITLE
GRF-342 category attributes handle issue in front related to the "unsaved changes" warning

### PR DESCRIPTION
**Pull request by FSAkeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/validation/category_commands.yml [hunk 1]
```diff
@@ -19,23 +19,20 @@ Akeneo\Category\Api\Command\UpsertCategoryCommand:
 Akeneo\Category\Command\UserIntents\SetText:
     properties:
       value:
-        - NotBlank: ~
```
Comment (FSAkeneo) [inline src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/validation/category_commands.yml:22]: Must not be required to be not blank, user can update an attribute value by resetting it to blank

File: src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx [hunk 2]
```diff
@@ -35,52 +38,38 @@ const FormContainer = styled.div`
...
-    [locale, onAttributeValueChange]
-  );
 
-  const handleImageChange = useCallback(
-    (attribute: Attribute) => (value: AttributeInputValue | null) => {
-      if (!isImageAttributeInputValue(value)) {
-        onAttributeValueChange(attribute, locale, null);
+      // attribute has textual type
+      const currentValue = getAttributeValue(attributeValues, attribute, locale);
+      if (attribute.type === CATEGORY_ATTRIBUTE_TYPE_RICHTEXT && !currentValue && value === '<p></p>\n') {
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx:55]: as it done in ReferenceEntity, you can declare a function in the model [like this](https://github.com/akeneo/pim-enterprise-dev/blob/78be0ddcc5d999cd3082e2cb995e7b92c4df6319/src/Akeneo/AssetManager/front/domain/model/asset/data/text.ts#L6-L7) and use it in the `handleChange`

Comment (FSAkeneo) [reply src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx:55]: it's not exactly the same, here we check the value before the change and dont wan't null as a valid value

Comment (FSAkeneo) [reply src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx]: ok it's refactored

File: src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/validation/category_commands.yml (hunk 1 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/validation/category_commands.yml:22]: its weird for a user to set a value to blank. if it's empty then the user's intention is not to delete it ?

**Review by BryanS29-akeneo — APPROVED** [review]

File: src/Akeneo/Category/back/Application/Converter/Checker/ValueCollectionRequirementChecker.php [hunk 3]
```diff
@@ -164,6 +164,6 @@ private static function assertImageData(?array $imageData): void
      */
     private static function assertTextData(string $textData): void
     {
-        Assert::stringNotEmpty($textData, "Expected String and not empty value for 'data'");
+        Assert::nullOrString($textData, "Expected String or null value for 'data'");
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/Application/Converter/Checker/ValueCollectionRequirementChecker.php:167]:
```suggestion
        Assert::string($textData, "Expected String or null value for 'data'");
```
The null value is reserved only when we have array and it is manage here : https://github.com/akeneo/pim-community-dev/blob/451cd9600d5497a009dc3116f6ecdc01692cb017/src/Akeneo/Category/back/Application/Converter/Checker/ValueCollectionRequirementChecker.php#L129. Functionaly, we cannot have a text set to null. This behavior will be manage when we will implemente the Value object in ValueCollection.

**Review by antoinetrouve — APPROVED** [review]

File: src/Akeneo/Category/back/Application/Converter/Checker/ValueCollectionRequirementChecker.php (hunk 3 above)
Comment (FSAkeneo) [reply src/Akeneo/Category/back/Application/Converter/Checker/ValueCollectionRequirementChecker.php:167]: True @antoinetrouve , thanks and well done, I make the changes

**Pull request merged**